ts_src/psetv2/input.ts

```diff
@@ -638,6 +689,48 @@ export class PsetInput {
       throw new Error('Invalid sighash type');
     }
 
+    if (this.explicitValue && !this.explicitValueProof) {
+      throw new Error(
+        'Explicit value proof is required if explicit value is set',
+      );
+    }
+
+    if (this.explicitValueProof && !this.explicitValue) {
+      throw new Error('Explicit value proof set without explicit value');
+    }
+
+    if (this.explicitAsset && !this.explicitAssetProof) {
+      throw new Error('Explicit asset proof is required if explicit asset set');
+    }
+
+    if (this.explicitAssetProof && !this.explicitAsset) {
+      throw new Error('Explicit asset proof set without explicit asset');
+    }
+
+    if (this.explicitAsset) {
+      const asset = AssetHash.fromBytes(this.explicitAsset);
+      if (asset.isConfidential()) {
+        throw new Error(`Explicit asset should be unconfidential`);
+      }
+    }
+
+    const utxo = this.getUtxo();
+    if (utxo && this.explicitAsset) {
+      if (!AssetHash.fromBytes(utxo.asset).isConfidential()) {
+        throw new Error(
+          'Explicit asset must be undefined if previous utxo is unconfidential',
+        );
+      }
+    }
+
+    if (utxo && this.explicitValue !== undefined) {
+      if (!ElementsValue.fromBytes(utxo.value).isConfidential()) {
+        throw new Error(
+          'Explicit value must be undefined if previout utxo is unconfidential',
+        );
+      }
+    }
```

Comment:
remove these checks. Explicit asset and value (and related proof) are not mandatory fields.